pkg/cloud/vsphere/services/kubeconfig/kubeconfig.go

```diff
@@ -43,18 +43,21 @@ func New(clusterName, controlPlaneEndpoint string, caKeyPair v1alpha1.KeyPair) (
 		return "", errors.New("key not found in status")
 	}
 
-	controlPlaneEndpointURL, err := url.Parse(controlPlaneEndpoint)
-	if err != nil {
-		if controlPlaneEndpointURL, err = url.Parse("https://" + controlPlaneEndpoint); err != nil {
+	var server string
+	// https://github.com/golang/go/blob/master/src/net/url/url.go#L469
+	if !strings.Contains(controlPlaneEndpoint, "://") {
+		server = strings.Split(controlPlaneEndpoint, "/")[0]
```

Comment:
Hi @pickledrick,

Should `SplitN` be used here?

Reply:
Happy to update, also added test cases for deeper paths.